lesson20/src/js/index.js

```diff
@@ -0,0 +1,93 @@
+import { createAttributedElements } from './utiles/createAttributeWithEllement';
+import { showLoadingImage, removeLoading } from './modules/loading';
+
+function displayErrorMassage(error) {
+  const displayArea = document.getElementById("js-table-contents");
+  const errorMassage = createAttributedElements("p", {class:"error-message"},error);
+  displayArea.appendChild(errorMassage);
+}
+
+const userItemName = {
+  userID : "ID",
+  userName : "名前",
+  userGender : "性別",
+  userAge : "年齢"
+}
+
+function renderTable(tableValue) {
+  const tableArea = document.getElementById('js-table-contents');
+  const userTable = createAttributedElements("table",{id:"js-user-table",class:"user-table"});
+  const tbody = document.createElement("tbody");
+  tbody.appendChild(createTableItem());
+  tbody.appendChild(createTableValue(tableValue));
+  tableArea.appendChild(userTable).appendChild(tbody);
+}
+
+function createTableItem (){
+  const frag = document.createDocumentFragment();
+  const tableRow = document.createElement("tr");
+  Object.keys(userItemName).forEach((key)=>{
+    const createItem = createAttributedElements("th",{class:"user-table__header"},userItemName[key]);
+    tableRow.appendChild(createItem);
+  });
+  frag.appendChild(tableRow);
+  return frag;
+}
+
+function createTableValue(userLists){
+  const userItemNameKey = Object.keys(userItemName);
+  const frag = document.createDocumentFragment();
+  userLists.forEach((user)=>{
+    const tableRow = document.createElement("tr");
+    userItemNameKey.forEach((key)=>{
+      for(let userKey in user){
+        if(userKey === key){
+          const tableValue = createAttributedElements("td",{class:"user-table__value"},user[userKey]);
+          frag.appendChild(tableRow).appendChild(tableValue);
+        }
+      }
+    })
```

Comment:
IMO
I don't think it would break the bank to write it this way.
```suggestion
  userLists.forEach((user)=>{
    const tableRow = document.createElement("tr");
    userItemNameKey.forEach((key)=>{
          const tableValue = createAttributedElements("td",{class:"user-table__value"},user[key]);
          frag.appendChild(tableRow).appendChild(tableValue);
    })
```

It looks like it would be good to separate this partial function!

Reply:
Thank you for your review!
Since userLists and userItemNameKey have the same key, I don't have to bother to check if the keys are the same!

I hadn't thought that far.Thank you!!!!!

[a263c7e](https://github.com/nyattsu72/hanson-frontend-cource/pull/28/commits/a263c7e8395eaac96d50fb7f31cb7f3897264f76)

Reply:
> Also, the part to put the value of table is separated and made into another function.[ed962e1](https://github.com/nyattsu72/hanson-frontend-cource/pull/28/commits/ed962e14a0b1b294f494f653fa18191b8c7dd4c7)

上のコミット部分、ちひろさんのレビュー箇所とも重複していると思うので
そちら修正いただいてからまた拝見させて頂きたいです!すみません。

Reply:
This has been fixed.
I would appreciate it if you could confirm this code.

[5807669](https://github.com/nyattsu72/hanson-frontend-cource/pull/28/commits/5807669e653f1f1ac28f4fed39942a24e5a11450)